src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -417,35 +438,141 @@ static MvcCoreLoggerExtensions()
                 11,
                 "List of registered output formatters, in the following order: {OutputFormatters}");
 
-            _ifMatchPreconditionFailed = LoggerMessage.Define<EntityTagHeaderValue>(
+            _writingRangeToBody = LoggerMessage.Define(
+                LogLevel.Debug,
+                17,
+                "Writing the requested range of bytes to the body...");
+
+            _registeredModelBinderProviders = LoggerMessage.Define<IModelBinderProvider[]>(
                 LogLevel.Debug,
                 12,
+                "Registered model binder providers, in the following order: {ModelBinderProviders}");
+
+            _attemptingToBindPropertyModel = LoggerMessage.Define<Type, string, Type, string>(
+               LogLevel.Debug,
+               13,
+               "Attempting to bind property '{PropertyContainerType}.{PropertyName}' of type '{ModelType}' using the name '{ModelName}' in request data ...");
+
+            _doneAttemptingToBindPropertyModel = LoggerMessage.Define<Type, string, Type>(
+               LogLevel.Debug,
+               14,
+               "Done attempting to bind property '{PropertyContainerType}.{PropertyName}' of type '{ModelType}'.");
+
+            _foundNoValueForPropertyInRequest = LoggerMessage.Define<string, Type, string, Type>(
+               LogLevel.Debug,
+               15,
+               "Could not find a value in the request with name '{ModelName}' for binding property '{PropertyContainerType}.{ModelFieldName}' of type '{ModelType}'.");
+
+            _foundNoValueInRequest = LoggerMessage.Define<string, string, Type>(
+               LogLevel.Debug,
+               16,
+               "Could not find a value in the request with name '{ModelName}' for binding parameter '{ModelFieldName}' of type '{ModelType}'.");
+
+            _noPublicSettableProperties = LoggerMessage.Define<string, Type>(
+               LogLevel.Debug,
+               17,
```

Comment:
What are these incrementing numbers for(17 in this case)?
What if tomorrow someone would need to add something in between, will they need to update all the numbers greater than the one they introduce?
If so, consider defining an integer on the top and referring to the same "i++" instead of specific numbers all over the place.
However, if these numbers are constants which shouldn't change, the proposal is not applicable.

Reply:
We never add anything in between, and using `i++` is dangerous because we might re-use numbers. The event IDs must remain identical across all versions of the product (changing it is a breaking change), so they must be constants (or literals).